src/Data/Bool/Compat.hs

```diff
@@ -12,6 +15,41 @@ module Data.Bool.Compat (
 import Data.Bool
 
 #if !MIN_VERSION_base(4,7,0)
+
+-- These instances are only valid if Bits isn't a subclass of Num (as Bool is
+-- not a Num instance), which is only true as of base-4.6.0.0 and later.
+# if MIN_VERSION_base(4,6,0)
+import Data.Bits (Bits(..))
+import Data.Eq ((/=))
+
+instance Bits Bool where
```

Comment:
Yeah, that sounds good.  We may eventually move all orphans into one central module that we import from all the other modules.